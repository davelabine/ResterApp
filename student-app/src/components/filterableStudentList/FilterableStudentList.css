
.filterableStudentList {
  padding: 40px 15px;
  margin-left: auto;
  margin-right: auto;
}

.buttons {
  text-align: right;
  margin: 1em 0;
}

.form {
  text-align: left;
}