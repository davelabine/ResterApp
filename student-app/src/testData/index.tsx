import { StudentData } from '../types';

export const STUDENT_DATA_EMPTY: StudentData = new StudentData();

export const STUDENT_DATA_BILLY: StudentData =  {
                                                    'skey': '8a80810d5dd89a3f015dd89ac8280000',
                                                    'id': 'Billy Bob',
                                                    'name': '1234',
                                                    'photo': {
                                                        'bucketName': 'resterapp-dev',
                                                        'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
                                                    }
                                                };

export const LIST_STUDENT_DATA_EMPTY: StudentData[] = [];

export const LIST_STUDENT_DATA_TWO: StudentData[] =  [
    {
        'skey': '8a80810d5dd89a3f015dd89ac8280000',
        'id': 'Billy Bob',
        'name': '1234',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a252800001',
        'id': '503074',
        'name': 'Albert Russell',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
];

export const LIST_STUDENT_DATA_BIG: StudentData[] =  [
    {
        'skey': '8a80810d5dd89a3f015dd89ac8280000',
        'id': '1234',
        'name': 'Billy Bob',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a252800001',
        'id': '503074',
        'name': 'Albert Russell',
        'photo': {
            'bucketName': 'resterapp-dev',
            'key': 'e93ecbf0-ba96-4290-a906-5ea67e4c4a9f'
        }
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a259b2000e',
        'id': '942226',
        'name': 'Alfredo Cummings',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a2677f0027',
        'id': '824388',
        'name': 'Allen Russell',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a271fb003a',
        'id': '266192',
        'name': 'Andre McKenna',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a27f0d0052',
        'id': '932083',
        'name': 'Angelica Obrien',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a281c80057',
        'id': '682307',
        'name': 'Anna Washington',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a27c52004d',
        'id': '734552',
        'name': 'Antonio Fowler',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a25fb80019',
        'id': '615643',
        'name': 'Ashley Mcdaniel',
        'photo': {}
    },
    {
        'skey': '8a80810d5dd89a3f015dd8a25897000c',
        'id': '753238',
        'name': 'Beatrice Russell',
        'photo': {}
    },
];
