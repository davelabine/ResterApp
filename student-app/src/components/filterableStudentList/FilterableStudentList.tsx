import * as React from 'react';
import './FilterableStudentList.css';

export interface Props {
  name: string;
}

class FilterableStudentList extends React.Component<Props, object> {
  render() {
    const { name } = this.props;

    return (
      <div className="filterableStudentList">
        <div className="well">
          <form className="form-inline">
              <label>Search Last Name: </label>
              <input type="text" name="name"/>
              <button type="button" className="btn btn-primary">Search</button>
          </form>
        </div>
        <table className="table table-striped table-sm table-hover">
          <thead>
            <tr>
              <th>Name</th>
              <th>ID</th>
              <th>URL</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>{name}</td>
              <td>12345</td>
              <td><a href="http://google.com">view</a> | 
              <a href="http://google.com"> edit</a></td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}
export default FilterableStudentList;

/*
<table class="table table-striped table-sm table-hover">
    <thead>
      <tr>
          <th>Name</th>
          <th>ID</th>
          <th>URL</th>
      </tr>
    </thead>
    <#list studentList as student>
    <tbody>
      <tr>
          <td>${student.name}</td>
          <td>${student.id}</td>
          <td><a id="viewStudent" href="${rootUrl}id/${student.skey}">view</a></td>
          <td><a id="editStudent" href="${rootUrl}id/${student.skey}/edit">edit</a></td>
      </tr>
      </#list>
    </tbody>
<#else>
  <div> No Students yet! </div>
</#if>
</table>
*/
