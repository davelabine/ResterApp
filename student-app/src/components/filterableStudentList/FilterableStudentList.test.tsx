import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { FilterableStudentList } from './FilterableStudentList';
import toJson from 'enzyme-to-json';
import * as studentTestData from '../../testData/';

describe('FilterableStudentList', () => {
    it('renders without crashing and matches the last snapshot', () => {
      const list = mount(
                      <FilterableStudentList 
                        students={studentTestData.LIST_STUDENT_DATA_TWO} 
                        filter=""
                        onFetchStudents={jest.fn()}
                        onAddStudent={jest.fn()}
                        onFilterStudents={jest.fn()}
                        onUpdateStudent={jest.fn()} 
                        onDeleteStudent={jest.fn()}
                      />);
      expect(toJson(list)).toMatchSnapshot();
    });

    it('calls onFetchStudents after being mounted', () => {
      const mockFetchStudents = jest.fn();
      shallow(
          <FilterableStudentList 
            students={studentTestData.LIST_STUDENT_DATA_TWO} 
            filter=""
            onFetchStudents={mockFetchStudents}
            onAddStudent={jest.fn()}
            onFilterStudents={jest.fn()}
            onUpdateStudent={jest.fn()} 
            onDeleteStudent={jest.fn()}
          />);
      expect(mockFetchStudents).toHaveBeenCalled();
    });
});
