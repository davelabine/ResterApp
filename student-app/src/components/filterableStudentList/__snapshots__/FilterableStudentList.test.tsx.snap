// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableStudentList renders without crashing and matches the last snapshot 1`] = `
<FilterableStudentList
  filter=""
  onAddStudent={[Function]}
  onDeleteStudent={[Function]}
  onFetchStudents={[Function]}
  onFilterStudents={[Function]}
  onUpdateStudent={[Function]}
  students={
    Array [
      Object {
        "id": "1234",
        "name": "Billy Bob",
        "photo": Object {
          "bucketName": "resterapp-dev",
          "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
        },
        "skey": "8a80810d5dd89a3f015dd89ac8280000",
      },
      Object {
        "id": "503074",
        "name": "Albert Russell",
        "photo": Object {
          "bucketName": "resterapp-dev",
          "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
        },
        "skey": "8a80810d5dd89a3f015dd8a252800001",
      },
    ]
  }
>
  <div
    className="filterableStudentList"
  >
    <Well
      bsClass="well"
    >
      <div
        className="well"
      >
        <StudentListFilterForm
          filter=""
          onAddStudent={[Function]}
          onFilterStudents={[Function]}
        >
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={true}
          >
            <form
              className="form-inline"
            >
              <FormGroup
                bsClass="form-group"
                controlId="formFilterStudents"
              >
                <div
                  className="form-group"
                >
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    <label
                      className="control-label"
                      htmlFor="formFilterStudents"
                    >
                      Search Last Name:
                    </label>
                  </ControlLabel>
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    type="text"
                    value=""
                  >
                    <input
                      className="form-control"
                      id="formFilterStudents"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </FormControl>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="pull-right"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="pull-right btn btn-default"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      Add Student
                    </button>
                  </Button>
                </div>
              </FormGroup>
              <EditStudentModal
                initialStudent={StudentData {}}
                onHide={[Function]}
                onSubmit={[Function]}
                show={false}
                submitButtonText="Add Student"
                title="Add Student"
              >
                <div
                  className="static-modal"
                >
                  <Modal
                    animation={true}
                    autoFocus={true}
                    backdrop={true}
                    bsClass="modal"
                    dialogClassName="custom-modal"
                    dialogComponentClass={[Function]}
                    enforceFocus={true}
                    keyboard={true}
                    manager={
                      ModalManager {
                        "add": [Function],
                        "containers": Array [],
                        "data": Array [],
                        "handleContainerOverflow": true,
                        "hideSiblingNodes": true,
                        "isTopModal": [Function],
                        "modals": Array [],
                        "remove": [Function],
                      }
                    }
                    onHide={[Function]}
                    renderBackdrop={[Function]}
                    restoreFocus={true}
                    show={false}
                  >
                    <Modal
                      autoFocus={true}
                      backdrop={true}
                      backdropClassName="modal-backdrop"
                      backdropTransitionTimeout={150}
                      containerClassName="modal-open"
                      dialogTransitionTimeout={300}
                      enforceFocus={true}
                      keyboard={true}
                      manager={
                        ModalManager {
                          "add": [Function],
                          "containers": Array [],
                          "data": Array [],
                          "handleContainerOverflow": true,
                          "hideSiblingNodes": true,
                          "isTopModal": [Function],
                          "modals": Array [],
                          "remove": [Function],
                        }
                      }
                      onEntering={[Function]}
                      onExited={[Function]}
                      onHide={[Function]}
                      renderBackdrop={[Function]}
                      restoreFocus={true}
                      show={false}
                      transition={[Function]}
                    />
                  </Modal>
                </div>
              </EditStudentModal>
            </form>
          </Form>
        </StudentListFilterForm>
        <StudentListItems
          filter=""
          onDeleteStudent={[Function]}
          onUpdateStudent={[Function]}
          students={
            Array [
              Object {
                "id": "1234",
                "name": "Billy Bob",
                "photo": Object {
                  "bucketName": "resterapp-dev",
                  "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                },
                "skey": "8a80810d5dd89a3f015dd89ac8280000",
              },
              Object {
                "id": "503074",
                "name": "Albert Russell",
                "photo": Object {
                  "bucketName": "resterapp-dev",
                  "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                },
                "skey": "8a80810d5dd89a3f015dd8a252800001",
              },
            ]
          }
        >
          <div>
            <Table
              bordered={false}
              bsClass="table"
              condensed={true}
              hover={true}
              responsive={false}
              striped={true}
            >
              <table
                className="table table-striped table-condensed table-hover"
              >
                <thead
                  className="studentListItemsHead"
                >
                  <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      ID
                    </th>
                    <th>
                      URL
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="studentListItemsBody"
                >
                  <StudentListItem
                    key="8a80810d5dd89a3f015dd89ac8280000"
                    onDeleteStudent={[Function]}
                    onUpdateStudent={[Function]}
                    student={
                      Object {
                        "id": "1234",
                        "name": "Billy Bob",
                        "photo": Object {
                          "bucketName": "resterapp-dev",
                          "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                        },
                        "skey": "8a80810d5dd89a3f015dd89ac8280000",
                      }
                    }
                  >
                    <tr>
                      <td>
                        Billy Bob
                      </td>
                      <td>
                        1234
                      </td>
                      <td>
                        <ButtonGroup
                          block={false}
                          bsClass="btn-group"
                          justified={false}
                          vertical={false}
                        >
                          <div
                            className="btn-group"
                          >
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              id="edit"
                              onClick={[Function]}
                            >
                              <button
                                className="btn btn-default"
                                disabled={false}
                                id="edit"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="pencil"
                                >
                                  <span
                                    className="glyphicon glyphicon-pencil"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              id="delete"
                              onClick={[Function]}
                            >
                              <button
                                className="btn btn-default"
                                disabled={false}
                                id="delete"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="trash"
                                >
                                  <span
                                    className="glyphicon glyphicon-trash"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                          </div>
                        </ButtonGroup>
                        <EditStudentModal
                          initialStudent={
                            Object {
                              "id": "1234",
                              "name": "Billy Bob",
                              "photo": Object {
                                "bucketName": "resterapp-dev",
                                "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                              },
                              "skey": "8a80810d5dd89a3f015dd89ac8280000",
                            }
                          }
                          onHide={[Function]}
                          onSubmit={[Function]}
                          show={false}
                          submitButtonText="Save"
                          title="Edit Billy Bob"
                        >
                          <div
                            className="static-modal"
                          >
                            <Modal
                              animation={true}
                              autoFocus={true}
                              backdrop={true}
                              bsClass="modal"
                              dialogClassName="custom-modal"
                              dialogComponentClass={[Function]}
                              enforceFocus={true}
                              keyboard={true}
                              manager={
                                ModalManager {
                                  "add": [Function],
                                  "containers": Array [],
                                  "data": Array [],
                                  "handleContainerOverflow": true,
                                  "hideSiblingNodes": true,
                                  "isTopModal": [Function],
                                  "modals": Array [],
                                  "remove": [Function],
                                }
                              }
                              onHide={[Function]}
                              renderBackdrop={[Function]}
                              restoreFocus={true}
                              show={false}
                            >
                              <Modal
                                autoFocus={true}
                                backdrop={true}
                                backdropClassName="modal-backdrop"
                                backdropTransitionTimeout={150}
                                containerClassName="modal-open"
                                dialogTransitionTimeout={300}
                                enforceFocus={true}
                                keyboard={true}
                                manager={
                                  ModalManager {
                                    "add": [Function],
                                    "containers": Array [],
                                    "data": Array [],
                                    "handleContainerOverflow": true,
                                    "hideSiblingNodes": true,
                                    "isTopModal": [Function],
                                    "modals": Array [],
                                    "remove": [Function],
                                  }
                                }
                                onEntering={[Function]}
                                onExited={[Function]}
                                onHide={[Function]}
                                renderBackdrop={[Function]}
                                restoreFocus={true}
                                show={false}
                                transition={[Function]}
                              />
                            </Modal>
                          </div>
                        </EditStudentModal>
                      </td>
                    </tr>
                  </StudentListItem>
                  <StudentListItem
                    key="8a80810d5dd89a3f015dd8a252800001"
                    onDeleteStudent={[Function]}
                    onUpdateStudent={[Function]}
                    student={
                      Object {
                        "id": "503074",
                        "name": "Albert Russell",
                        "photo": Object {
                          "bucketName": "resterapp-dev",
                          "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                        },
                        "skey": "8a80810d5dd89a3f015dd8a252800001",
                      }
                    }
                  >
                    <tr>
                      <td>
                        Albert Russell
                      </td>
                      <td>
                        503074
                      </td>
                      <td>
                        <ButtonGroup
                          block={false}
                          bsClass="btn-group"
                          justified={false}
                          vertical={false}
                        >
                          <div
                            className="btn-group"
                          >
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              id="edit"
                              onClick={[Function]}
                            >
                              <button
                                className="btn btn-default"
                                disabled={false}
                                id="edit"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="pencil"
                                >
                                  <span
                                    className="glyphicon glyphicon-pencil"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                            <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              id="delete"
                              onClick={[Function]}
                            >
                              <button
                                className="btn btn-default"
                                disabled={false}
                                id="delete"
                                onClick={[Function]}
                                type="button"
                              >
                                <Glyphicon
                                  bsClass="glyphicon"
                                  glyph="trash"
                                >
                                  <span
                                    className="glyphicon glyphicon-trash"
                                  />
                                </Glyphicon>
                              </button>
                            </Button>
                          </div>
                        </ButtonGroup>
                        <EditStudentModal
                          initialStudent={
                            Object {
                              "id": "503074",
                              "name": "Albert Russell",
                              "photo": Object {
                                "bucketName": "resterapp-dev",
                                "key": "e93ecbf0-ba96-4290-a906-5ea67e4c4a9f",
                              },
                              "skey": "8a80810d5dd89a3f015dd8a252800001",
                            }
                          }
                          onHide={[Function]}
                          onSubmit={[Function]}
                          show={false}
                          submitButtonText="Save"
                          title="Edit Albert Russell"
                        >
                          <div
                            className="static-modal"
                          >
                            <Modal
                              animation={true}
                              autoFocus={true}
                              backdrop={true}
                              bsClass="modal"
                              dialogClassName="custom-modal"
                              dialogComponentClass={[Function]}
                              enforceFocus={true}
                              keyboard={true}
                              manager={
                                ModalManager {
                                  "add": [Function],
                                  "containers": Array [],
                                  "data": Array [],
                                  "handleContainerOverflow": true,
                                  "hideSiblingNodes": true,
                                  "isTopModal": [Function],
                                  "modals": Array [],
                                  "remove": [Function],
                                }
                              }
                              onHide={[Function]}
                              renderBackdrop={[Function]}
                              restoreFocus={true}
                              show={false}
                            >
                              <Modal
                                autoFocus={true}
                                backdrop={true}
                                backdropClassName="modal-backdrop"
                                backdropTransitionTimeout={150}
                                containerClassName="modal-open"
                                dialogTransitionTimeout={300}
                                enforceFocus={true}
                                keyboard={true}
                                manager={
                                  ModalManager {
                                    "add": [Function],
                                    "containers": Array [],
                                    "data": Array [],
                                    "handleContainerOverflow": true,
                                    "hideSiblingNodes": true,
                                    "isTopModal": [Function],
                                    "modals": Array [],
                                    "remove": [Function],
                                  }
                                }
                                onEntering={[Function]}
                                onExited={[Function]}
                                onHide={[Function]}
                                renderBackdrop={[Function]}
                                restoreFocus={true}
                                show={false}
                                transition={[Function]}
                              />
                            </Modal>
                          </div>
                        </EditStudentModal>
                      </td>
                    </tr>
                  </StudentListItem>
                </tbody>
              </table>
            </Table>
          </div>
        </StudentListItems>
      </div>
    </Well>
  </div>
</FilterableStudentList>
`;
